orquesta/conducting.py

```diff
@@ -882,3 +885,36 @@ def get_task_transition_contexts(self, task_id, route):
                 contexts[task_transition_id] = self.get_task_initial_context(t[1], route)
 
         return contexts
+
+    def request_workflow_rerun(self, rerun_options):
+        # Check current workflow status.
+        if self.get_workflow_status() not in statuses.ABENDED_STATUSES:
+            LOG.debug('Workflow execution cannot rerun from task(s) because it is not in a '
+                      'failed state')
+            return
+
+        # Force reset workflow status to running
+        self.workflow_state.status = statuses.RUNNING
```

Comment:
Per discussion in our thread, set the workflow status to `states.RESUMING` and then let existing message plumbing in st2workflowengine to handle the `resuming` status change.